opentracing-util/src/test/java/io/opentracing/util/GlobalTracerTestUtil.java

```diff
@@ -1,67 +0,0 @@
-/*
- * Copyright 2016-2018 The OpenTracing Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
- * in compliance with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
- * or implied. See the License for the specific language governing permissions and limitations under
- * the License.
- */
-package io.opentracing.util;
-
-import io.opentracing.Tracer;
-import io.opentracing.noop.NoopTracerFactory;
-
-import java.lang.reflect.Field;
-
-/**
- * Utility class for manipulating the {@link GlobalTracer} when testing.
- * <p>
- * The {@linkplain GlobalTracer} has register-once semantics, but for testing purposes it is useful to
- * manipulate the globaltracer to guarantee certain preconditions.
- * <p>
- * The {@code GlobalTracerTestUtil} can be included in your own code by adding the following dependency:
- * <pre><code>
- *     &lt;dependency>
- *         &lt;groupId>io.opentracing&lt;/groupId>
- *         &lt;artifactId>opentracing-util&lt;/artifactId>
- *         &lt;version><em>version</em>&lt;/version>
- *         <strong>&lt;type>test-jar&lt;/type></strong>
- *         &lt;scope>test&lt;/scope>
- *      &lt;/dependency>
- * </code></pre>
- */
-public class GlobalTracerTestUtil {
```

Comment:
Removing this class breaks backward compatibility for folks using the `test-jar` dependency!

Reply:
I can revert this change, it's not essential to the fix. I should be able to update the isRegistered boolean in the same way the tracer is reset to a NoopTracer.

However, should we be supporting the use of GlobalTracerTestUtil to reset the tracer outside of the GlobalTracer jar?

Reply:
Instead of updating two variables, have you could considered a constant UNREGISTERED_TRACER and check with `==` if a tracer was registered?